python/utilities.py

```diff
@@ -162,6 +162,40 @@ def getDesiredHeadingDegrees(position, localWaypoint):
     return math.degrees(math.atan2(xy[1], xy[0]))
 
 
+def getLOSHeadingDegrees(position, previousWaypoint, currentWaypoint):
+    '''Calculate the heading that the boat should aim towards to reach the local waypoint.
+    The LOS guidance law will attempt to keep the boat on the straight line between the previous and current waypoint.
+
+    Args:
+       position (sailbot_msg.msg._latlon.latlon): Current position of the boat
+       previousWaypoint (sailbot_msg.msg._latlon.latlon): Previous local waypoint that the boat visited
+       currentWaypoint (sailbot_msg.msg._latlon.latlon): Current local waypoint that the boat is aiming towards
+
+    Returns:
+       float that is the heading (degrees) that the boat should be aiming towards to reach the local waypoint
+    '''
+    # The lookahead distance describes how "hard" the boat should try to reduce the crosstrack error
+    lookahead = 0.001
+    Kp = 1 / lookahead
+
+    x_start = previousWaypoint.lon
```

Comment:
Code is clean and easy to understand. 

One thing to note is that the units of lon and lat are not the same length, so angle calculation may not be right. Could use latlonToXY to calculate in km space.

Reply:
ah, of course! Totally forgot about the lon/lat length